toffy/mph_comp_test.py

```diff
@@ -0,0 +1,145 @@
+import pandas as pd
+import pytest
+import os
+import tempfile
+import json
+
+from pathlib import Path
+
+from toffy import mph_comp as mph
+
+
+def test_get_estimated_time():
+    bad_path = os.path.join(Path(__file__).parent, "data", "not-a-folder")
+    bad_fov = "not-a-fov"
+
+    good_path = os.path.join(Path(__file__).parent, "data", "tissue")
+    good_fov = 'fov-1-scan-1'
+
+    # bad directory path should raise an error
+    with pytest.raises(ValueError):
+        mph.get_estimated_time(bad_path, good_fov)
+
+    # bad fov name data should raise an error
+    with pytest.raises(FileNotFoundError):
+        mph.get_estimated_time(good_path, bad_fov)
+
+    # bad FOV json file data should raise an error, no frameSize or dwellTimeMillis keys
+    bad_data = {"fov": {"not_frameSize": 0, "not_dwellTimeMillis": 0}}
+    temp_json = tempfile.NamedTemporaryFile(mode="w", suffix='fov_name.json', delete=False)
+    json.dump(bad_data, temp_json)
+    json_dir = tempfile.gettempdir()
+    with pytest.raises(KeyError, match="missing one of the necessary keys"):
+        mph.get_estimated_time(json_dir, 'fov_name')
```

Comment:
`tempfile.NamedTemporaryFile` can be kind of wonky.  Best to just use tempdir usually:
```python
with tempfile.TemporaryDirectory() as tmpdir:
    with open(os.path.join(tmpdir, 'fov_name.json'), 'w') as f:
        json.dump(bad_data, f)
    with pytest.raises(...):
        mph.get_estimated_time(tmpdir, 'fov_name')
```